src/crypto/crypto_cipher.cc

```diff
@@ -27,7 +27,8 @@ using v8::Value;
 namespace crypto {
 namespace {
 bool IsSupportedAuthenticatedMode(const EVP_CIPHER* cipher) {
-  switch (EVP_CIPHER_mode(cipher)) {
+  const int mode = EVP_CIPHER_mode(cipher);
+  switch (mode) {
   case EVP_CIPH_CCM_MODE:
```

Comment:
Assigning an `unsigned long` (OpenSSL) or `uint32_t` (BoringSSL) to an `int` is also a narrowing conversion.

Unfortunately, there is no `OPENSSL_NO_CCM`, and there is also no (basic) signed integer type that is large enough to hold all possible return values of `EVP_CIPHER_mode`.

I was going to suggest `OPENSSL_IS_BORINGSSL`, but then the code will break if BoringSSL ever adds support for CCM, I guess.

So maybe some explicit condition like this?

```suggestion
  switch (EVP_CIPHER_mode(cipher)) {
#if !(EVP_CIPH_CCM_MODE < 0)  // BoringSSL workaround.
  case EVP_CIPH_CCM_MODE:
#endif
```

Reply:
this still doesn't compile in BoringSSL:

```
../../third_party/electron_node/src/crypto/crypto_cipher.cc:34:8: error: case value evaluates to -1, which cannot be narrowed to type 'uint32_t' (aka 'unsigned int') [-Wc++11-narrowing]
  case EVP_CIPH_CCM_MODE:
       ^
../../third_party/boringssl/src/include/openssl/cipher.h:510:27: note: expanded from macro 'EVP_CIPH_CCM_MODE'
#define EVP_CIPH_CCM_MODE (-1)
                          ^
1 error generated.
ninja: build stopped: subcommand failed.
```

Reply:
I'll double-check if there was a particular reason we made them negative, but that the constants, even if never used, don't fit in the EVP_CIPHER_mode return value is kinda silly of us. I can just fix them to be positive. That seems easier all around. :-)

Reply:
https://boringssl-review.googlesource.com/c/boringssl/+/57025

Reply:
Change to make them positive is now in BoringSSL! Hopefully that removes the need for Node to hack around this?

Reply:
Patch confirmed to resolve build issue with no changes in Node.js! Closing this out, thank you ❤️

Reply:
I think that makes sense.